src/libopensc/iso7816.c

```diff
@@ -213,6 +213,9 @@ int decode_do_data(struct sc_context *ctx,
 			sc_asn1_decode(ctx, asn1_do_data, encoded_data, encoded_data_len, NULL, NULL),
 			"sc_asn1_decode() failed");
 
+	if (!(asn1_do_data[1].flags & SC_ASN1_PRESENT))
+		return SC_ERROR_INVALID_ASN1_OBJECT;
+
```

Comment:
From the commit message and change I do not see how this is solving the issue. You mark the field optional, but if it is not present, you fail here?

Reply:
When receiving the data from the card, it is mandatory, because we're expecting data. This is what is tested here. Additionally, we use the same prototype for encoding the request, where we don't include data (just the offset). That's why the ASN.1 object is *optional* in the prototype.